### PR TITLE
Support input and stream condition

### PR DESCRIPTION
**Pull request by nchaulet**
## Description 

Related to https://github.com/elastic/package-spec/issues/86  and https://github.com/elastic/integrations/pull/411

@ycombinator added the support for `condition` at the input and stream level for this change to be available to agent it need to be available in the package registry so Kibana can see it.

This PR expose the `condition` property both at the input level and the stream level

Should we had test data?  I tested with the system package from https://github.com/elastic/integrations/pull/411 and it work as expected

### CODE REVIEW

**Comment by ruflin** [comment]
It would be good to add a test package with an example of this in https://github.com/elastic/package-registry/tree/master/testdata/package if possible

This PR will need a changelog entry.

**Review by mtojek — CHANGES_REQUESTED** [review]
As @ruflin already stated, please try to add the `condition` field to some test packages, so we can see it present in HTTP responses. To regenerate expected test output use the `-generate` switch (`go test . -generate`).

**Comment by nchaulet** [comment]
@elasticmachine merge upstream

**Comment by mtojek** [comment]
> @elasticmachine merge upstream

I'm not quite convinced that it works here :) used to perform it manually.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,8 +13,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
-* Add "traces" as legal event type. [#656](https://github.com/elastic/package-registry/pull/656)
-* Add input-level `template_path` field. [#655](https://github.com/elastic/package-registry/pull/655)
+- Add "traces" as legal event type. [#656](https://github.com/elastic/package-registry/pull/656)
```
Comment (mtojek) [inline CHANGELOG.md:16]: Out of curiosity - why did you refactored the CHANGELOG :) ? I'm not complaining, but maybe you can push it in a separate PR? /cc @ruflin

File: testdata/generated/search-all.json [hunk 2]
```diff
@@ -69,6 +69,16 @@
     "download": "/epr/input_level_templates/input_level_templates-1.0.0.zip",
     "path": "/package/input_level_templates/1.0.0"
   },
+  {
+    "name": "input_stream_condition",
+    "title": "Input and stream condtion package",
```
Comment (mtojek) [inline testdata/generated/search-all.json:74]: nit: typo

**Review by mtojek — APPROVED** [review]
I'm wondering if the package-registry shouldn't filter package list based on the condition (just a thought).

LGTM, although let's double-check it with @ycombinator or @ruflin .

File: CHANGELOG.md (hunk 1 above)
Comment (nchaulet) [reply CHANGELOG.md:16]: Oh sorry I did not check, I think my editor format on save did this, I will revert this

**Comment by nchaulet** [comment]
While working on this PR I am wondering if we really need a new variables for input condition or if we could put this in the agent template, I started the discussion here I would love to have other thoughts on that https://github.com/elastic/package-spec/issues/86#issuecomment-740152447

**Comment by nchaulet** [comment]
Closed as we will be using agent template for condition as discussed here https://github.com/elastic/package-spec/issues/86

**Pull request closed without merging**